src/Microsoft.Tye.Hosting/ProxyExtensions.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
```

Comment:
This comes from https://github.com/aspnet/AspLabs/blob/85ff16588fe970d738aa51ff3ba1c675bc678f28/src/Proxy/src/ProxyAdvancedExtensions.cs